Expensify/App
# Pin reports feature

<Assign reviewers that know the area or changes well. Feel free to tag any additional reviewers you see fit.>

<Explanation of the change or anything fishy that is going on>

### Fixed Issues
Fixes https://github.com/Expensify/Expensify/issues/141697

### Tests
1. Open Expensify chat. Search for a person to DM and click on the star next to the name to pin the room. 
1. Pin two or three more DMs.
1. Confirm the UI shows up like so,
UI web, (pinned)
![image](https://user-images.githubusercontent.com/14356145/96312014-96511b00-0fbf-11eb-88f9-0090b668f2c3.png)

UI web (unpinned)
![image](https://user-images.githubusercontent.com/14356145/96311854-3e1a1900-0fbf-11eb-85af-91aaa877f179.png)

UI Mobile (pinned),
![image](https://user-images.githubusercontent.com/14356145/96311899-5853f700-0fbf-11eb-8a9a-c41391ec7475.png)

UI Mobile (unpinned),
![image](https://user-images.githubusercontent.com/14356145/96311925-64d84f80-0fbf-11eb-92fb-c2f0d6af086c.png)

4. The sorting order is by pinned followed by unpinned.
All rooms pinned shows me this order,
![image](https://user-images.githubusercontent.com/14356145/96311952-7588c580-0fbf-11eb-83ef-a4586e10e03b.png)

If Alberto was not pinned i get,
![image](https://user-images.githubusercontent.com/14356145/96311972-7b7ea680-0fbf-11eb-8413-e80dff3b81cb.png)

### Screenshots
Shown above.
